src/vendor/Path.ts

```diff
@@ -340,9 +340,21 @@ export default class Path {
     return Deno.readTextFile(this.string)
   }
 
-  readLines(): AsyncIterableIterator<string> {
+  async *readLines(): AsyncIterableIterator<string> {
     const fd = Deno.openSync(this.string)
-    return readLines(fd)
+    const lines = readLines(fd)
+    try {
+      while (true) {
+        const res = await lines.next()
+        if (res.done) {
+          break
+        }
+        yield res.value
+      }
+    }
```

Comment:
Does this need to re-throw if there's an error? And would this work more simply?
```typescript
for await(const line of lines) {
  yield line.value
}
fd.close()
```
or will it actually throw?

Reply:
yeah that's much better, I even did it that way in the test code.  And the try/finally will allow the error to continue up the stack while correctly freeing the file resource, so catch and rethrow is no necessary.